src/renderer/components/alert/SwapMightBeCancelledAlert.tsx

```diff
@@ -14,24 +17,24 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 export default function SwapMightBeCancelledAlert({
-  dbState,
   bobBtcLockTxConfirmations,
 }: {
-  dbState: MergedDbState;
   bobBtcLockTxConfirmations: number;
 }) {
   const classes = useStyles();
+  const swap = useActiveSwapInfo();
 
-  const timelockStatus = getTimelockStatus(
-    dbState.state.Bob.ExecutionSetupDone.state2.cancel_timelock,
-    dbState.state.Bob.ExecutionSetupDone.state2.punish_timelock,
-    bobBtcLockTxConfirmations
-  );
-
-  if (bobBtcLockTxConfirmations < 3) {
+  if (
+    bobBtcLockTxConfirmations < 5 ||
+    swap === null ||
+    swap.timelock === null
+  ) {
     return <></>;
```

Comment:
**Suggestion:** Consider adding error handling for the case when `swap.timelock` is `null`. Currently, the code simply returns an empty fragment when `swap.timelock` is `null`. It would be more user-friendly to display a message indicating that the timelock information could not be retrieved.
```suggestion
  if (
    bobBtcLockTxConfirmations < 5 ||
    swap === null ||
    swap.timelock === null
  ) {
    return <Alert severity="error">Could not retrieve timelock information</Alert>;
  }
```